resolwe/flow/execution_engines/workflow/__init__.py

```diff
@@ -143,9 +143,12 @@ def evaluate(self, data):
             # Copy permissions.
             copy_permissions(data, data_object)
 
-            # Copy collections.
-            for collection in data.collection_set.all():
-                collection.data.add(data_object)
+            # Copy collection.
+            if not data_object.collection and data.collection:
+                # Typically collection will be set during data object
+                # creation. Only if not set there, inherit from workflow.
+                data_object.collection = data.collection
```

Comment:
Why don't we pass this in `Data.objects.create` in line `131`? We can avoid another query to DB.